src/gluonts/dataset/repository/_m1.py

```diff
@@ -0,0 +1,102 @@
+import json
+import os
+from pathlib import Path
+from typing import NamedTuple, List, Dict
+from urllib import request
+from zipfile import ZipFile
+
+from gluonts.dataset.repository._util import metadata, to_dict, save_to_file
+
+from ._tsf_reader import TSFReader
+
+ROOT = "https://zenodo.org/record"
+
+dataset_info = {
+    "m1_yearly": {"file": "m1_yearly_dataset.zip", "record": "4656193"},
+    "m1_quarterly": {"file": "m1_quarterly_dataset.zip", "record": "4656154"},
+    "m1_monthly": {"file": "m1_monthly_dataset.zip", "record": "4656159"},
+}
+
+
+def download_dataset(description: str, path: Path):
+    request.urlretrieve(
+        f"{ROOT}/{description['record']}/files/" f"{description['file']}",
+        path / description["file"],
+        )
+
+
+def frequency_converter(freq: str):
+    split = freq.split("_")
+    if len(split) == 1:
+        return freq[0].upper()
+    elif len(split) == 2 and split[0].isnumeric():
+        return f"{split[0]}{split[1].upper()}"
+    else:
+        raise Exception("Unknown frequency")
+
+
+def save_metadata(
+        dataset_path: Path, cardinality: int, freq: str, prediction_length: int
+):
+    with open(dataset_path / "metadata.json", "w") as f:
+        f.write(
+            json.dumps(
+                metadata(
+                    cardinality=cardinality,
+                    freq=freq,
+                    prediction_length=prediction_length,
+                )
+            )
+        )
+
+
+def save_dataset(
+        dataset_path: Path, data: List[Dict], prediction_length: int = None
```

Comment:
The name implies that you just save something, why should the `prediction_length` matter then? Looks like this should be a two-step thing.

Reply:
Yes, you are right, it was a bit too hacky here. It's now split